### ISSUE TITLE
Pass rule.deep-exit on TestMain function

### ISSUE THREAD

**Issue by bpeake-illuscio** [mgechev/revive#290]
**Describe the bug**
When `[rule.deep-exit]` is being enforced, calling `os.Exit()` in the `TestMain()` function is flagged as an error. 

The official go documentation for testing explicitly makes a call to `os.Exit()` at the end of `TestMain()` (see below), so it seems intuitive that it should be added to the whitelist of functions allowed to call os.Exit() with the rule active.

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
# flags

revive -config revive.toml ./...
```

```toml
[rule.deep-exit]
```

3. Have a TestMain function in tests which calls os.Exit() for custom failure conditions. In this example we are declaring a configurable minimum code coverage to pass tests:

```
// This function handles running of all the tests.
func TestMain(m *testing.M) {
	// call flag.Parse() here if TestMain uses flags
	minCoverageFlag := flag.Float64(
		"minimum-coverage",
		 // Default to 85% coverage requirement.
		0.85,
		"minimum coverage for passing tests from 0.0 (none) - 1.0 (all lines)",
	)

	flag.Parse()

	// Run the tests
	testResults := m.Run()

	// testResults 0 means we've passed,
	// and CoverMode will be non empty if run with -cover
	if testResults == 0 && testing.CoverMode() != "" {
		coverageResult := testing.Coverage()
		if coverageResult < *minCoverageFlag {
			_, _ = fmt.Println(
				"Tests passed but coverageResult of '", coverageResult, "' does " +
					"not meet minimum requirement of '", *minCoverageFlag, "'",
			)
			testResults = -1
		}
	}

	// Return with exit code.
	os.Exit(testResults)
}
```

**Expected behavior**
`[rule.deep-exit]` should not `flag os.Exit()` here, since `TestMain()` acts as a top-level entry point for tests. Quoting: https://golang.org/pkg/testing/

> It is sometimes necessary for a test program to do extra setup or teardown before or after testing. It is also sometimes necessary for a test to control which code runs on the main thread. To support these and other cases, if a test file contains a function:
> 
> ```func TestMain(m *testing.M)```
>
> then the generated test will call TestMain(m) instead of running the tests directly. TestMain runs in the main goroutine and can do whatever setup and teardown is necessary around a call to m.Run. It should then call os.Exit with the result of m.Run. When TestMain is called, flag.Parse has not been run. If TestMain depends on command-line flags, including those of the testing package, it should call flag.Parse explicitly.
> 
> A simple implementation of TestMain is:
> 
> ```
> func TestMain(m *testing.M) {
> 	// call flag.Parse() here if TestMain uses flags
> 	os.Exit(m.Run())
> }
> ```

**Desktop (please complete the following information):**
 - OS: OSX Mojave 10.14.1
 - Go version 1.13.5

**Other Thoughts**

Obviously it's easy to disable the rule for this function, so not a high priority issue, but might be worth adding to the linter logic.

**Comment by chavacava** [comment]
Hi @bpeake-illuscio, thanks for filling the issue. Please check if the PR #294 behaves as you expect.
